# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1750709642 -7200
# Parent  81f706d03a532ae91018d63b6f922069c221316c
Bug 1973294 - Ensure header view delete button removes message, not attachment.
Inspired by https://phabricator.services.mozilla.com/D254603?vs=on&id=1066967 (early version, better one to come)

diff --git a/mail/base/content/msgHdrView.js b/mail/base/content/msgHdrView.js
--- a/mail/base/content/msgHdrView.js
+++ b/mail/base/content/msgHdrView.js
@@ -3721,13 +3721,16 @@ function MsgComposeDraftMessage() {
 }
 
 const trashButtonClickHandler = event => {
   if (event.button == 0) {
+    const prevFocusedElement = Services.focus.focusedElement;
+    getMessagePaneBrowser().focus();
     goDoCommand(
       event.shiftKey && event.target.dataset.imapDeleted == "false"
         ? "cmd_shiftDelete"
         : "cmd_delete"
     );
+    prevFocusedElement.focus();
   }
 };
 
 /**
